packages/driver/src/cypress/server.coffee

```diff
@@ -272,6 +272,7 @@ create = (options = {}) ->
         testStr(fullyQualifiedUrl, options.stripOrigin(fullyQualifiedUrl))
 
     xhrMatchesRoute: (xhr, route) ->
+      return route.matcher(xhr, route) if _.isFunction(route.matcher)
```

Comment:
Can you add tests for the desired behavior?

The specs for cy.server, cy.route, etc. are all here: https://github.com/cypress-io/cypress/blob/develop/packages/driver/test/cypress/integration/commands/xhr_spec.coffee